extmod/uasyncio/taskgroup.py

```diff
@@ -0,0 +1,244 @@
+# Adapted with permission from the EdgeDB project.
```

Comment:
This will need a MicroPython copyright and license header, like the other uasyncio source files.